.github/workflows/ubuntu-windows.yml

```diff
@@ -0,0 +1,401 @@
+name: Windows-Ubuntu
+
+on:
+  push:
+    branches:
+      - main
+      - feature/*
+  release:
+    types: [ created ]
+
+env:
+  GITHUB_TOKEN: ${{ github.token }}
+
+jobs:
+
+  build_sirius:
+    name: ${{ matrix.os }} build sirius
+    runs-on: ${{ matrix.os }}
+    env:
+      Sirius_Ref: antares_integration
+    strategy:
+      matrix:
+        os: [windows-2019, ubuntu-20.04]
+    steps:
+      - name: Get release
+        if: github.event_name == 'release' && github.event.action == 'created'
+        id: get_release
+        uses:
+          bruceadams/get-release@v1.2.3
+
+      - name: install zip for windows
+        if: ${{ startsWith(matrix.os, 'windows') }}
+        shell: cmd
+        run: |
+          choco install zip --no-progress
+
+      - name: Checkout or-tools
+        uses: actions/checkout@v2
+
+      - name: setup sirius
+        id: setup
+        uses: ./.github/workflows/build-sirius
+        with:
+          sirius-ref: ${{ env.Sirius_Ref }}
+
+      - name: Upload sirius install
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.os }}_sirius
+          path: ${{ steps.setup.outputs.sirius-install-path }}
+
+      - name: Publish sirius install
+        if:  ${{ github.event_name == 'release' && github.event.action == 'created' }}
+        uses: actions/upload-release-asset@v1.0.2
+        with:
+          upload_url: ${{ steps.get_release.outputs.upload_url }}
+          asset_path: ${{ steps.setup.outputs.sirius-archive-path }}
+          asset_name: ${{ matrix.os }}_${{ steps.setup.outputs.sirius-archive-name }}
+          asset_content_type: application/zip
+
+
+  build:
+    needs: [build_sirius]
+    name: ${{ matrix.os }} shrd=${{ matrix.shared }} sirius=${{ matrix.sirius }} xprs=${{ matrix.xprs }} extras=${{ matrix.extras }} java=${{ matrix.java }} dotnet=${{ matrix.dotnet }} python=${{ matrix.python }}-${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    env:
+      Sirius_Ref: antares_integration
+      XPRESSDIR: ${{ github.workspace }}/xpressmp813
+      XPRESS: ${{ github.workspace }}/xpressmp813/bin
+      XPRS_LIB_Path: ${{ github.workspace }}/xpressmp813/lib
+      SIRIUS_INSTALL_PATH : ${{ github.workspace }}/sirius_install
+      SIRIUS_BIN_DIR: ${{ github.workspace }}/sirius_install/bin
+      XPRESS_BIN_DIR: ${{ github.workspace }}/xpressmp813/bin
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-2019, ubuntu-20.04]
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+        python: [ON, OFF]
+        xprs: [ON]
+        sirius: [ON, OFF]
+        shared: [ON, OFF]
+        extras: [ON, OFF]
+
+        exclude:
+          - xprs: OFF
+            sirius: OFF
+          - os: windows-2019
+            shared: ON
+          - os: windows-2019
+            shared: OFF
```

Comment:
Question : if shared == ON, only the shared library is built, and if shared == OFF, only the static lib is build ?

Reply:
Short Answer: yes
My problem was that we need to build different type of libraries for different OS.
We need static C++-libraries for both Windows and linux (for Antares)
Under windows we can only build ortools Python/Java/C# libraries using a STATIC build
Under Linux we can only build ortools Python/Java/C# libraries using a SHARED build

So we build
Windows only static with the support to all lagnuages (Extras=ON)
Linux static only with the C++ support (Extras=OFF)
Lunix shared with support to all languages (Extras=ON)

Should we change the matrix ?

Reply:
No, maybe add a comment here

Reply:
done